aiootp/commons/slots.py

```diff
@@ -60,19 +60,27 @@ def __init_subclass__(cls, /, *a: t.Any, **kw: t.Any) -> None:
             for subcls in cls.__mro__
             for name in getattr(subcls, "__slots__", ())
         })
+        cls._slots_set = frozenset(cls.__slots__)
 
     def __init__(
         self, mapping: t.Mapping[t.Hashable, t.Any] = {}, /, **kw: t.Any
     ) -> None:
         """
-        Maps the user-defined kwargs to the instance attributes. If a
-        subclass defines a `__slots__` list, then only variables with
-        names in the list can be admitted to the instance. Defining
-        classes with `__slots__` can greatly increase memory efficiency
-        if a system instantiates many objects of the class.
-        """
-        for name, value in {**dict(mapping), **kw}.items():  # flexible. subclasses
-            self[name] = value                               # should prefer specific
+        Maps user-defined kwargs to instance attributes. If a subclass
+        defines a `__slots__`, then only the names declared within can
+        be admitted to the instance. Unless `"__dict__"` is also added.
+        Defining classes with slots improves a system's memory efficiency,
+        especially when many instances are created. Having an instance
+        dictionary offers flexibility, though the interplay with slots
+        can cause problems. This initializer avoids setting the names
+        declared in `__slots__` within the a potential instance dict.
+        """
```

Comment:
This isn't visible enough in just the init. Consider expanding on the validity table from the PR, & adding it to at least the module doc of ``namespaces.py``, and the setattr that's shared by the class hierarchy.

ie.
### Validity of assignment styles:
```python
✔ hybrid.attr = "value"                # valid
✔ hybrid["attr"] = "value"             # valid
✔ setattr(hybrid, "attr", "value")     # valid

🚫 hybrid.__dict__["attr"] = "value"    # invalid
```

Reply:
Banner added to ([aiootp/commons/slots.py][slots], [aiootp/commons/typed_slots.py][typed_slots], [aiootp/commons/namespaces.py][namespaces])

https://github.com/rmlibre/aiootp/blob/fd652586117e90c5f87fd8c4e2a28a55ede17425/aiootp/commons/namespaces.py#L35-L55

Resolved by (https://github.com/rmlibre/aiootp/pull/11/commits/bc98732b68fe660ece1024c4591cd28d3f9d1bee)

[slots]: https://github.com/rmlibre/aiootp/blob/bc98732b68fe660ece1024c4591cd28d3f9d1bee/aiootp/commons/slots.py
[typed_slots]: https://github.com/rmlibre/aiootp/blob/bc98732b68fe660ece1024c4591cd28d3f9d1bee/aiootp/commons/typed_slots.py
[namespaces]: https://github.com/rmlibre/aiootp/blob/bc98732b68fe660ece1024c4591cd28d3f9d1bee/aiootp/commons/namespaces.py